Source/WebInspectorUI/UserInterface/Views/TreeOutline.js

```diff
@@ -402,12 +402,21 @@ WI.TreeOutline = class TreeOutline extends WI.Object
 
     _forgetTreeElementAndDescendants(element)
     {
+        if (element === this.selectedTreeElement) {
+            const suppressNotification = true;
+            element.deselect(suppressNotification);
+        }
```

Comment:
are we sure we need both parts?  im pretty sure this bit in particular always happens in every caller, so there shouldn't be a need to do it again (tho i think it might be worth having a `console.assert(!element.selected, element);` just in case)

Reply:
Having both parts is how we make sure the original behavior was restored. When I only had the part at the top and not this part for each child, the bug persisted.